lib/rubrowser/data.rb

```diff
@@ -1,30 +1,74 @@
 require 'rubrowser/parser/factory'
+require 'tsort'
 
 module Rubrowser
   class Data
+    attr_reader :definitions, :relations
+
     def initialize(files)
       @files = files
-      @parsed = false
       parse
     end
 
-    def definitions
-      @_definitions ||= parsers.map(&:definitions).reduce(:+).to_a
+    def mark_circular_dependencies
+      mark_circular_components(components)
+    end
+
+    def mark_circular_components(components)
+      @definitions.each do |definition|
+        if components.include?(definition.namespace.first.to_s)
+          definition.set_circular
+        end
+      end
+
+      @relations.each do |relation|
+        if components.include?(relation.namespace.to_s)
```

Comment:
In this case, I feel that using &&/|| for control flow would make the code unclear. How do you feel?